### PR TITLE
Update Thrift protocol to use ECS fields

### PR DESCRIPTION
**Pull request by andrewkroh**
That dashboard was updated too.

Here's a summary of what fields changed.

Part of #7968

Changed

- bytes_in -> source.bytes
- bytes_out -> destination.bytes
- notes -> error.message
- responsetime -> event.duration (unit are now nanoseconds)

Added

- source
- destination
- event.dataset = thrift
- event.end
- event.start
- network.community_id
- network.transport = tcp
- network.protocol = thrift
- network.bytes
- network.type

Unchanged Packetbeat Fields

- method
- path
- query
- status
- type = thrift (we might remove this since we have event.dataset)

### CODE REVIEW

File: packetbeat/protos/thrift/thrift.go [hunk 1]
```diff
@@ -1130,22 +1137,15 @@ func (thrift *thriftPlugin) publishTransactions() {
 						t.reply.exceptions)
 				}
 			}
-			if len(t.reply.notes) > 0 {
-				fields["notes"] = t.reply.notes
+			if len(t.reply.notes) == 1 {
```
Comment (ruflin) [inline packetbeat/protos/thrift/thrift.go]: This becomes a pattern across protocols :-)

File: packetbeat/tests/system/test_0002_thrift_basics.py [hunk 2]
```diff
@@ -7,13 +7,13 @@ def tutorial_asserts(self, objs):
...
         # all except the zip async calls.
-        assert all([o["bytes_in"] > 0 for o in objs])
-        assert all([o["bytes_out"] > 0 for o in objs[0:14]])
-        assert all([o["bytes_out"] > 0 for o in objs[16:]])
-        assert objs[14]["bytes_out"] == 0
-        assert objs[15]["bytes_out"] == 0
+        assert all([o["source.bytes"] > 0 for o in objs])
+        assert all([o["destination.bytes"] > 0 for o in objs[0:14]])
+        assert all([o["destination.bytes"] > 0 for o in objs[16:]])
+        #assert objs[14]["destination.bytes"] == 0
```
Comment (ruflin) [inline packetbeat/tests/system/test_0002_thrift_basics.py]: Commeted out on purpose? Also 2 more below.

Comment (andrewkroh) [reply packetbeat/tests/system/test_0002_thrift_basics.py]: Zero values are automatically omitted based on how I marshaling these fields. Effectively it's like `json:"destination.bytes,omitempty"`. I meant to remove these lines. Will fix.

**Comment by andrewkroh** [comment]
> Packetbeat doesn't use Ingest Node, correct?

That's correct. And Packetbeat is using `error.message` to indicate internal PB errors/warnings like some packets were missed rather than errors responses in the protocol like invalid sql query.

**Pull request merged**